stdlib/dynarray.ml

```diff
@@ -217,36 +211,36 @@ let shrink_capacity v : unit =
 let iter k v =
   let n = v.size in
   for i = 0 to n-1 do
-    k (Array.unsafe_get v.arr i)
+    k (Array.get v.arr i)
```

Comment:
I was rather considering something safely using `unsafe_get` as follows:
```ocaml
let iter k v =
  let n = v.size in
  let arr = v.arr in
  if n > Array.length arr then raise ConcurrentModification ;
  for i = 0 to n-1 do
    let x = Array.unsafe_get arr i in
    if is_null x then raise ConcurrentModification ;
    k x
  done
```
(where `is_null` is always false in your version but should be checked in a version with null checks.)

Idem for other iterators.

Reply:
I think it's very reasonable, but I'll re-add them after we get a good implementation for `null`, so that I can really think through each case.

When the time comes, should `ConcurrentModification` be added to `Dynarray`, or directly as one of the main exceptions in `Stdlib`? On could imagine `Hashtbl` starting to use it too, if it's helpful.